ntoskrnl/CMakeLists.txt

```diff
@@ -44,9 +44,17 @@ endif()
 
 target_link_libraries(ntoskrnl cportlib csq ${PSEH_LIB} arbiter cmlib ntlsalib rtl ${ROSSYM_LIB} libcntpr wdmguid ioevent)
 
+# dynamic analysis switches
 if(STACK_PROTECTOR)
     target_link_libraries(ntoskrnl gcc_ssp)
-elseif(RUNTIME_CHECKS)
+endif()
+
+if(SANITIZE_UB)
+    target_compile_options(ntoskrnl PUBLIC "-fsanitize=undefined;-fno-sanitize=alignment")
```

Comment:
Are you using PUBLIC here for any particular reason? The description makes it sound like that'll apply to anyone linking to ntoskrnl/hal.